client/src/components/overview/Overview.jsx

```diff
@@ -5,22 +5,240 @@ import AddToCart from './parts/addToCart/AddToCart.jsx';
 import ImageGallery from './parts/ImageGallery/ImageGallery.jsx';
 import axios from 'axios';
 
+class Overview extends React.Component {
+  constructor(props) {
+    super(props);
 
+    this.state = {
+      data: [],
+      SKU: null,
+      expanded: false,
+      styles: [],
+      current: [],
+      mainIndex: 0,
+      amount: 0,
+      currentThumbnails: [],
+      currentStyle: null,
+      thumbnailSection: 0,
+      selectedSize: null
+    }
 
-function Overview() {
+    this.mainSlide = this.mainSlide.bind(this);
+    this.updateMainPic = this.updateMainPic.bind(this);
+    this.handleExpand = this.handleExpand.bind(this);
+    this.updateStyle = this.updateStyle.bind(this);
+    this.updateThumbnailSection = this.updateThumbnailSection.bind(this);
+    this.selectSize = this.selectSize.bind(this);
+    this.checkThumbnailSection = this.checkThumbnailSection.bind(this);
+  }
 
+  updateStyle(style) {
+    this.setState({
+      currentStyle: style,
+      mainIndex: 0,
 
-  const [info, getInfo] = useState([]);
+    })
 
+    // console.log(this.state.currentStyle)
+  }
 
-  return (
-    <div>
-      <ProductInfo />
-      <StyleSelector />
-      <AddToCart />
-      <ImageGallery />
-    </div>
-  )
+  updateThumbnailSection(dir) {
+
+    if (this.state.thumbnailSection === 0 && dir === -1) {
+      return;
+    }
+    if (this.state.thumbnailSection === 1 && dir === 1) {
+      return;
+    }
+
+    let newIndex = this.state.currentThumbnails[this.state.thumbnailSection + dir][0].index;
+
+    console.log('newIndex', newIndex)
+
+
+    this.setState({
+      thumbnailSection: this.state.thumbnailSection + dir,
+      mainIndex: newIndex
+    })
+
+  }
+
+
+  checkThumbnailSection(dir) {
+    let thumbnailLength = this.state.currentThumbnails[0].length;
+
+    if ((this.state.mainIndex + dir === this.state.amount)) {
+      return;
+    }
+
+    if (this.state.mainIndex + dir > thumbnailLength - 1) {
+
+      this.setState({
+        thumbnailSection: 1
+      })
+    }
+
+    if (this.state.mainIndex + dir <= thumbnailLength - 1) {
+
+      this.setState({
+        thumbnailSection: 0
+      })
+    }
+
+  }
+
+  handleExpand() {
+    this.setState({
+      expanded: !this.state.expanded
+    })
+  }
+
+  // handles left or right main button click
+  mainSlide(dir) {
+
+    if (this.state.mainIndex === 0 && dir === -1) {
+      return
+    }
+
+    if (this.state.mainIndex === this.state.amount - 1 && dir > 0) {
+      return
+    }
+
+    if (this.state.mainIndex === 0) {
+      this.setState({
+        current: this.state.styles.results[0].photos[dir],
+        mainIndex: this.state.mainIndex + dir
+      })
+
+    } else {
+      this.setState({
+        current: this.state.styles.results[0].photos[this.state.mainIndex + dir],
```

Comment:
Consider using the development server, the endpoint for your widget to clean up how the data is sent to the front-end.

Instead of a pass through of styles{results: [{photos}]}, for this data point you could just send [{photos}]

Also, for deeply nested properties on collections be safe! Use [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining)